packages/query-core/src/queryObserver.ts

```diff
@@ -395,8 +355,7 @@ export class QueryObserver<
     if (
       isServer ||
       resolveEnabled(this.options.enabled, this.#currentQuery) === false ||
-      !isValidTimeout(this.#currentRefetchInterval) ||
-      this.#currentRefetchInterval === 0
+      !isValidTimeout(this.#currentRefetchInterval)
```

Comment:
note: timeout 0 is now not valid per default, so we can remove the extra check here for `refetchInterval`, too.